apps/mobile/src/screens/Onboarding/OnboardingUsernameScreen.tsx

```diff
@@ -82,41 +97,77 @@ export function OnboardingUsernameScreen() {
     setUsername(text);
   }, []);
 
+  const [isCreatingUser, setIsCreatingUser] = useState(false);
   const handleNext = useCallback(async () => {
-    setUsernameError('');
-
     try {
       // If its existing user, update the username
       if (user?.username) {
         await updateUser(user.dbid, username, user.bio ?? '');
 
         navigation.navigate('OnboardingProfileBio');
+
         return;
       }
 
+      setIsCreatingUser(true);
       const response = await createUser(authMechanism, username, bio);
 
       if (response.createUser?.__typename === 'CreateUserPayload') {
-        // TODO: Remove this once incremental sync is implemented
-        if (!isSyncing) {
-          syncTokens('Ethereum');
+        // If the user is signing up with email, redirect to the bio screen
+        if (authMethod === 'Email') {
+          navigation.navigate('OnboardingProfileBio');
+          return;
         }
 
-        navigation.navigate('OnboardingProfileBio');
+        // If the user is signing up with a wallet, attached the email to the user
+        if (authMethod === 'Wallet' && userEmail) {
+          await updateEmail(userEmail);
+        }
+
+        const user = response.createUser.viewer?.user;
+        const ensAddress = user?.potentialEnsProfileImage?.wallet?.chainAddress;
+
+        // 1. if the the dont have ens pfp, go to the pfp screen
+        // 2. if they do, set an ENS and go to the profile bio screen
+        if (ensAddress) {
+          await setEnsProfileImage({
+            address: ensAddress.address,
+            chain: ensAddress.chain,
+          });
+          if (!isSyncing) {
+            syncTokens('Ethereum');
+          }
+          navigation.navigate('OnboardingProfileBio');
+        } else {
+          if (!isSyncing) {
+            await syncTokens('Ethereum');
```

Comment:
let's add: `Zora` and `Base`. not a high priority, i think we can do this as a follow-up for both mobile and web sign-ups.

Reply:
Added `Zora` and `Base` too!